File: src/com/jda/functionalprograms/Quadratic.java
package com.jda.functionalprograms;

import com.jda.utility.Maths;
import com.jda.utility.Maths.ComplexNumber;
import com.jda.utility.Reader;
import java.util.HashSet;

public
class Quadratic {

  public static
  void main(String[] args) {
    System.out.println("Let's find the roots of a Quadratic equation a*x*x + b*x + c = 0");
    Reader reader = new Reader();
    System.out.println("Please enter a b c");
    double a = reader.readDouble();
    double b = reader.readDouble();
    double c = reader.readDouble();
    HashSet<ComplexNumber> roots = Maths.findRoots(a, b, c);
    System.out.println((roots.size() > 1) ? "The roots are " : "The root is ");
    roots.forEach(System.out::println);
  }
}

File: src/com/jda/functionalprograms/Quadratic.kt
package com.jda.functionalprograms

import com.jda.utility.Utility
import com.jda.utility.Utility.ComplexNumber
import java.util.function.Consumer

object Quadratic {

    @JvmStatic
    fun main(args: Array<String>) {
        println("Let's find the roots of a Quadratic equation a*x*x + b*x + c = 0")
        val reader = Utility().Reader()
        println("Please enter a b c")
        val a = reader.readDouble()
        val b = reader.readDouble()
        val c = reader.readDouble()
        val roots = Utility.findRoots(a, b, c)
        println(if (roots.size > 1) "The roots are " else "The root is ")
        roots.forEach(Consumer<ComplexNumber> { println(it) })
    }
}
